ttnn/cpp/ttnn/operations/pool/maxpool/max_pool.hpp

```diff
@@ -173,3 +179,46 @@ Tensor maxpool2d_new(const Tensor &input,
 
 }  // namespace tt_metal
 }  // namespace tt
+
+namespace ttnn::operations {
+namespace maxpool {
+
+using array2_t = std::array<uint32_t, 2>;
```

Comment:
array of 2 is a pair, right? can we use a pair instead of an array?
can we remove this from header?
or can we give it a distinct name, instead of array2? 
I am not sure if this is better `using Dillation = .. `

Reply:
It's used for other args like `kernel_size`, `stride`, and `padding` so not sure `Dilation` would be appropriate here. I think best approach is to refactor all of these args (including `batch_size`, input_h`, `input_w`, `channels`) into an op config struct.

Reply:
I'm not following, what is the issue with using array?